src/compiler/crystal/interpreter/compiler.cr

```diff
@@ -176,6 +176,7 @@ class Crystal::Repl::Compiler < Crystal::Visitor
     # And we have to do them starting from the end because it's a stack.
     node.args.reverse_each do |arg|
       block_var = node.vars.not_nil![arg.name]
+      next if block_var.name == "_"
```

Comment:
Unused variables can also be given a make as long as it starts with `_` (e g. `_var2`), so should this check for the first character being `_` as opposed to the entire string? Same comment for other places.

Reply:
The sole underscore is the only unused variable at the language level, you straight up cannot read from it, unlike those other prefixed variables.